map-downloader-android/app/src/main/java/com/here/android/example/map/downloader/MainActivity.java

```diff
@@ -18,48 +18,55 @@
 
 import android.Manifest;
 import android.app.ListActivity;
+import android.content.Context;
 import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.v4.app.ActivityCompat;
 import android.view.View;
 import android.widget.ListView;
 import android.widget.Toast;
 
-import java.util.ArrayList;
-import java.util.List;
-
 /**
  * Main activity which launches the map list view and handles Android run-time requesting
  * permission.
  */
-
 public class MainActivity extends ListActivity {
-    private final static int REQUEST_CODE_ASK_PERMISSIONS = 1;
+    private static final int REQUEST_CODE_ASK_PERMISSIONS = 1;
+    private static final String[] RUNTIME_PERMISSIONS = {
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.INTERNET,
+            Manifest.permission.ACCESS_WIFI_STATE,
+            Manifest.permission.ACCESS_NETWORK_STATE};
```

Comment:
Style: I believe we usually put `};` into its own line and indent the block only with 4 spaces, in this case.